Documentation/security/network/encryption-ztunnel.rst

```diff
@@ -0,0 +1,262 @@
+2. Cluster Mesh is not enabled (ztunnel is currently not compatible with Cluster Mesh).
+
+.. tabs::
+
+    .. group-tab:: Cilium CLI
+
+       If you are deploying Cilium with the Cilium CLI, pass the following
+       options:
+
+       .. parsed-literal::
+
+          cilium install |CHART_VERSION| \
+             --set encryption.enabled=true \
+             --set encryption.type=ztunnel
+
+    .. group-tab:: Helm
+
+       If you are deploying Cilium with Helm by following
+       :ref:`k8s_install_helm`, pass the following options:
+
+       .. parsed-literal::
+
+           helm install cilium |CHART_RELEASE| \\
+             --namespace kube-system \\
+             --set encryption.enabled=true \\
+             --set encryption.type=ztunnel
+
+ztunnel may also be enabled manually by setting the ``enable-ztunnel: true``
+option in the Cilium ``ConfigMap`` and restarting each Cilium agent instance.
+
+.. note::
+
+   By default, ztunnel listens on the Unix socket at ``/var/run/cilium/ztunnel.sock``.
+   If your ztunnel installation uses a different socket path, you can configure it
+   using the ``--set ztunnel.zdsUnixAddr=/path/to/ztunnel.sock`` Helm option.
+
+Enrolling Namespaces
+====================
+
+After enabling ztunnel in Cilium, you need to explicitly enroll namespaces to
+enable mTLS encryption for their workloads. This is done by applying a label
+to the namespace:
+
+.. code-block:: shell-session
+
+    kubectl label namespace <namespace-name> mtls-enabled=true
+
+To verify that a namespace is enrolled:
+
+.. code-block:: shell-session
+
+    kubectl get namespace <namespace-name> --show-labels
+
+When a namespace is enrolled:
+
+1. All existing pods in the namespace (except ztunnel pods themselves) are enrolled
+2. Iptables rules are configured in each pod's network namespace for traffic redirection
+3. Pod metadata is sent to the ztunnel proxy via the ZDS protocol
+4. Future pods created in the namespace are automatically enrolled
+
+To disenroll a namespace:
+
+.. code-block:: shell-session
+
+    kubectl label namespace <namespace-name> mtls-enabled-
+
+This will:
+
+1. Disenroll all pods in the namespace from ztunnel
+2. Remove the iptables rules from each pod's network namespace
+3. Notify ztunnel to stop processing traffic for those workloads
+
+Validate the Setup
+==================
+
+1. Check that ztunnel has been enabled:
+
+   .. code-block:: shell-session
+
+      kubectl -n kube-system describe cm cilium-config | grep enable-ztunnel -A2
+
+   You should see output indicating that ztunnel encryption is enabled.
+
+2. Check which namespaces are enrolled:
+
+   .. code-block:: shell-session
+
+      kubectl get namespaces -l mtls-enabled=true
+
+   This shows all namespaces labeled for ztunnel enrollment.
+
+   To verify that these namespaces are actually enrolled in the StateDB table:
+
+   .. code-block:: shell-session
+
+      kubectl exec -n kube-system ds/cilium -- cilium-dbg statedb dump
```

Comment:
Won't this dump a lot of state, including state not related to ztunnel? Is it viable to include example output here to help a user to see what "successful" output of this command looks like?

Reply:
Perhaps an intermediate solution might be to pipe the output into `jq ...` with an expression that filters the content for the ztunnel-related tables?